easybuild/framework/easyblock.py

```diff
@@ -1345,10 +1345,16 @@ def make_module_extend_modpath(self):
             # add user-specific module path; use statement will be guarded so no need to create the directories
             user_modpath = build_option('subdir_user_modules')
             if user_modpath:
+                user_envvars = build_option('envvars_user_modules') or ['HOME']
                 user_modpath_exts = ActiveMNS().det_user_modpath_extensions(self.cfg)
                 self.log.debug("Including user module path extensions returned by naming scheme: %s", user_modpath_exts)
-                txt += self.module_generator.use(user_modpath_exts, prefix=self.module_generator.getenv_cmd('HOME'),
-                                                 guarded=True, user_modpath=user_modpath)
+                for user_envvar in user_envvars:
+                    if not os.getenv(user_envvar):
+                        raise EasyBuildError("Requested environment variable %s as an additional branch for user"
```

Comment:
Use `$` when referring to environment modules, add missing space after `user`

```suggestion
                        raise EasyBuildError("Requested environment variable $%s as an additional branch for user "

Reply:
Actually, do we really need to be this restrictive?
Why does the environment variable need to be defined at this point?

It only matters when the module file is consumed, and whatever the value of the environment variable is here, it doesn't say much about the value it could have when the module is loaded.

Reply:
No strictly not, I was trying to think of a way to hammer home that the environment variable must exist or the module file will be broken.

Reply:
I wanted to guard checking the environment variable and that is really easy in Lua
```
os.getenv("ENVVAR") or ""
```
in fact, in our use case, if the variable doesn't exist the return value is `nil` and this is equivalent since we are calling this in a guarded statement that is using `pathJoin`.

The problem is with Tcl modules, in this case if the  envvar does not exist you get an error like:
```
alanc@~/EasyBuild_Git/easybuild-framework(alt-envvar-user-mod)$ module show test2
Lmod has detected the following error:  /home/alanc/EasyBuild_Git/EB_Devel/test2: (test2): can't read "::env(HOME2)": no such variable 
While processing the following module(s):
    Module fullname  Module Filename
    ---------------  ---------------
    test2            /home/alanc/EasyBuild_Git/EB_Devel/test2
```
This is because we are trying to reference the value in an array when we use `$env::("HOME2")`. I tried to put a guard around this but my Tcl-foo is not making it work

Reply:
Ok, I found a Tcl syntax that can give the same behaviour as Lua:
```
[if { [info exists ::env(HOME)] } { concat $::env(HOME) } ]
```